test/RouteResultTest.php

```diff
@@ -45,7 +46,7 @@ public function testRouteFailureRetrieveAllHttpMethods()
     public function testRouteFailureRetrieveHttpMethods()
     {
         $result = RouteResult::fromRouteFailure();
-        $this->assertSame([], $result->getAllowedMethods());
+        $this->assertSame(['*'], $result->getAllowedMethods());
```

Comment:
Behaviour here has been changed, we can set default value to `[]` and typehint on `?array` in `fromRouteFailure` method to keep the previous functionality.
@weierophinney What do you think?

I'd keep it consistent.

Reply:
We need to make sure that the consumers (the individual routing libraries) are able to upgrade without issue. So long as the same argument types are accepted, how the default works is less important.

That said, `[]` seems VERY DIFFERENT than `['*']`; the first looks like an empty set (no HTTP methods specified), while the latter looks like it accepts a "wildcard" (i.e., any method).

I think in this case, we need to look at the `RouteMiddleware` in Expressive to determine what it expects.

Reply:
I can't see any usages of `fromRouteFailure` method in Expressive.

Reply:
Look in the individual routers, then; just looked, and the FastRouter bridge definitely calls it.